sqld/src/lib.rs

```diff
@@ -228,6 +237,53 @@ fn check_fresh_db(path: &Path) -> bool {
     !path.join("wallog").exists()
 }
 
+fn validate_extensions(extensions_path: Option<PathBuf>) -> anyhow::Result<Vec<PathBuf>> {
+    let mut valid_extensions = vec![];
+    if let Some(ext_dir) = extensions_path {
+        let extensions_list = ext_dir.join("trusted.lst");
+        let mut list = File::open(&extensions_list).with_context(|| {
+            format!(
+                "While trying to read extensions: cannot load {}",
+                &extensions_list.display()
+            )
+        })?; // anyhow;
+        let mut file_contents = String::new();
+
+        list.read_to_string(&mut file_contents)
+            .with_context(|| format!("can't read {}", &extensions_list.display()))?;
+
+        let extensions = file_contents.split('\n').filter(|x| !x.is_empty());
+
+        for line in extensions {
+            let ext_info: Vec<&str> = line.trim().split(' ').filter(|x| !x.is_empty()).collect();
+
+            let ext_sha = ext_info.first().ok_or_else(|| {
+                anyhow::anyhow!("invalid line on {}: {}", &extensions_list.display(), line)
+            })?;
+            let ext_fname = ext_info.get(1).ok_or_else(|| {
+                anyhow::anyhow!("invalid line on {}: {}", &extensions_list.display(), line)
+            })?;
```

Comment:
We can use `str.split_once()` here, something like `let (ext_sha, ext_fname) = line.trim().split_once(|c| c.is_whitespace()).with_context(...)`

Reply:
split_once and other method act on bytes and not char: that be surprising sometimes.

Reply:
Ok, I thought `|c| c.is_whitespace()` would match a stretch of whitespace characters, but you are right, it matches just a single character. However, it does work on characters, not on bytes (it handles UTF-8 correctly; after all, this is Rust! :))